pkg/server/server_test.go

```diff
@@ -540,6 +539,85 @@ var _ = Describe("ApplicationsServer", func() {
 		})
 	})
 
+	Describe("AddApplication", func() {
+		It("adds an app with an unspecified config repo", func() {
+			ctx := context.Background()
+			name := "my-app"
+			appRequest := &pb.AddApplicationRequest{
+				Name:      name,
+				Namespace: namespace.Name,
+				Url:       "ssh://git@github.com/some-org/somerepo.git",
+				Path:      "./k8s/mydir",
+				Branch:    "main",
+			}
+			gp.GetRepoVisibilityStub = func(s string) (*gitprovider.RepositoryVisibility, error) {
+				return gitprovider.RepositoryVisibilityVar(gitprovider.RepositoryVisibilityInternal), nil
+			}
+
+			gp.CreatePullRequestToUserRepoStub = func(urr gitprovider.UserRepositoryRef, s1, s2 string, cf []gitprovider.CommitFile, s3, s4, s5 string) (gitprovider.PullRequest, error) {
```

Comment:
For just setting up the return you could use this syntax:

`gp.CreatePullRequestToUserRepoReturns(testutils.DummyPullRequest{}, nil)`

It should help cleaning up a bit all those mocks